tools/runner

```diff
@@ -209,6 +209,12 @@ for key in libs.keys():
 
 test_params['defines'] = test_params['defines'].split()
 
+# Filter test files based on what the runner claims to support.
+test_params['files'] = [
+    f for f in test_params['files']
+    if os.path.splitext(f)[1] in runner_obj.allowed_extensions
+]
+
```

Comment:
Could we change it to a regular for so that we can show a `self.info` message or maybe a warning when skipping a file? It works fine now but may make our lives easier if something is unexpectedly skipped in the future